### PR TITLE
[node] Update typings to v22.6.0

### PR DESCRIPTION
**Pull request by Renegade334**
https://github.com/nodejs/node/releases/tag/v22.6.0

- inspector: add initial support for network inspection
- stream: expose DuplexPair API
- test_runner: add globPatterns option
- test_runner: add context.filePath

Supersedes #70528.

----

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `pnpm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

If changing an existing definition:

- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/nodejs/node/blob/main/doc/changelogs/CHANGELOG_V22.md#22.6.0
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the `package.json`.

### CODE REVIEW

File: types/node/test.d.ts [hunk 1]
```diff
@@ -528,6 +535,12 @@ declare module "node:test" {
          * @param message Message to be reported.
          */
         diagnostic(message: string): void;
+        /**
+         * The absolute path of the test file that created the current test. If a test file imports
+         * additional modules that generate tests, the imported tests will return the path of the root test file.
+         * @since v22.6.0
+         */
+        readonly filePath: string | undefined;
```
Comment (Renegade334) [inline types/node/test.d.ts:543]: This property is `undefined` if there is no path associated with the test, _eg._ if running in REPL.

File: types/node/test.d.ts [hunk 2]
```diff
@@ -758,6 +771,12 @@ declare module "node:test" {
      * @since v18.7.0, v16.17.0
      */
     class SuiteContext {
+        /**
+         * The absolute path of the test file that created the current suite. If a test file imports
+         * additional modules that generate suites, the imported suites will return the path of the root test file.
+         * @since v22.6.0
+         */
+        readonly filePath: string | undefined;
```
Comment (Renegade334) [inline types/node/test.d.ts:779]: Ditto.

**Comment by DangerBotOSS** [comment]
## Formatting

The following files are not formatted:
1. types/node/test/inspector.ts
1. types/node/test/stream.ts

Consider running `pnpm dprint fmt` on these files to make review easier.
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 83637c17ac3e82aca8e7e770f4cf8d599095fc18
</p>

**Review by jakebailey — APPROVED** [review]

**Comment by Renegade334** [comment]
Ready to merge.

**Pull request merged**